### PR TITLE
Add numpy to typing deps

### PR DESCRIPTION
**Pull request by dstansby**
Adding numpy, numba, and dask to the environment in which type checking is done revealed some more fixes that needed making. Builds on top of https://github.com/brainglobe/cellfinder-core/pull/161

### CODE REVIEW

File: src/cellfinder_core/detect/filters/volume/ball_filter.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def append(self, plane: np.ndarray, mask: np.ndarray) -> None:
                 [e for e in plane.shape[:2]],
             )
             assert [e for e in mask.shape[:2]] == [
-                e for e in self.inside_brain_tiles.shape[2]
+                e for e in self.inside_brain_tiles.shape[:2]
```
Comment (dstansby) [inline src/cellfinder_core/detect/filters/volume/ball_filter.py:144]: This is a bugfix that typing noticed!

File: src/cellfinder_core/detect/filters/volume/volume_filter.py [hunk 2]
```diff
@@ -66,7 +66,7 @@ def __init__(
         )
 
         self.cell_detector = get_cell_detector(
-            plane_shape=self.setup_params[0].shape,
+            plane_shape=self.setup_params[0].shape,  # type: ignore
```
Comment (dstansby) [inline src/cellfinder_core/detect/filters/volume/volume_filter.py:69]: `mypy` doesn't know the dimensions of the array here, so gives `self.setup_params[0].shape` a type `Tuple[int, ...]` (no known length), but `plane_shape` expectes `Tuple[int, int` (length 2). I wasn't sure how to fix this, so ignored it for now.

Comment (alessandrofelder) [reply src/cellfinder_core/detect/filters/volume/volume_filter.py:69]: OK!

I'll open a low-priority issue for when we want to think more about types or want to refactor the code around `get_cell_detector` :shrug:

**Review by alessandrofelder — APPROVED** [review]
Great, thanks! LGTM.

**Pull request merged**